From 917904f93ea90ed9e6471bfe6d6f784d82e010ac Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Tue, 14 Jul 2020 10:25:57 -0300
Subject: [PATCH] Add Makefile option to disable initial mouse grab.

Specify WITH_MOUSE_GRAB=false or when the debug target is being built.

Updated from Olof-Joachim Frahm <olof@macrolet.net> Debian patch
---
 Makefile       | 7 ++++++-
 src/keyboard.c | 2 +-
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 658a5ef..ca2fc08 100644
--- a/Makefile
+++ b/Makefile
@@ -9,6 +9,7 @@ else
 endif
 
 WITH_NETWORK := true
+WITH_GRAB_MOUSE := true
 
 ################################################################################
 
@@ -60,6 +61,10 @@ ifneq ($(OPENTYRIAN_VERSION), )
     EXTRA_CPPFLAGS += -DOPENTYRIAN_VERSION='"$(OPENTYRIAN_VERSION)"'
 endif
 
+ifeq ($(WITH_GRAB_MOUSE), true)
+    EXTRA_CPPFLAGS += -DWITH_GRAB_MOUSE
+endif
+
 CPPFLAGS := -DNDEBUG
 CFLAGS := -pedantic
 CFLAGS += -MMD
@@ -99,7 +104,7 @@ ALL_LDLIBS = -lm \
 all : $(TARGET)
 
 .PHONY : debug
-debug : CPPFLAGS += -UNDEBUG
+debug : CPPFLAGS += -UNDEBUG -UWITH_GRAB_MOUSE
 debug : CFLAGS += -Werror
 debug : CFLAGS += -O0
 debug : CFLAGS += -g3
diff --git a/src/keyboard.c b/src/keyboard.c
index 82a2936..d63ae8c 100644
--- a/src/keyboard.c
+++ b/src/keyboard.c
@@ -43,7 +43,7 @@ Uint8 keysactive[SDL_NUM_SCANCODES];
 bool new_text;
 char last_text[SDL_TEXTINPUTEVENT_TEXT_SIZE];
 
-#ifdef NDEBUG
+#ifdef WITH_GRAB_MOUSE
 bool input_grab_enabled = true;
 #else
 bool input_grab_enabled = false;
-- 
2.26.2

